### PR TITLE
Add max capacity effects for locked weapons

### LINKED ISSUE #1386: Dragon Tooth weapons have 0 strength.
### Environment
* **FreeOrion Version**: 0.4.6+
* **Operating System**: Fedora 25
* **Graphic card used**: N/A
* **Fetched as** 
  * Compiled from source cf67d52

### Description
When awarded a Dragon Tooth from an Ancient Ruins, the ship has a total weapon capacity of 0.

Seen with empires without Death Ray researched, possibly due to the change in default weapon effects.

### PR DESCRIPTION
**Pull request by dbenage-cx**
Fixes #1386 

Grants ~a reduced~ damage and shot value for weapons that are not yet researched by an empire, but owned by some means.

### CODE REVIEW

**Comment by Dilvish-fo** [comment]
I like the general structure of the fix but I am not sure about the reduction.  Am I reading this right that it would allow the weapon to operate, but with a 20% reduction to the base attack value?  Could you compare the numbers for the Dragon Tooth, which is our primary motivation for this?

I guess this would also apply to ships acquired via Psy Dom, and while the 20% reduction would seem reasonable in that case, I don't think it's really needed, and it really doesn't seem to be what we want for the Dragon Tooth; if we keep them in the same bucket via this mechanism it seems it should be at base value, not 80% of base.

**Comment by dbenage-cx** [comment]
Ok, have removed the reduction.
That was primarily to address a suspicion that the issue affects gifted ships, but I've not tested to verify.

I suppose an alternate method might be to add a starting tech with such effects (possibly hidden as unresearchable).

File: default/scripting/ship_parts/Weapons.focs.txt [hunk 1]
```diff
@@ -147,3 +157,18 @@ WEAPON_BASE_UNOWNED_EFFECTS
...
+                Not Unowned
+                Not OwnerHasShipPartAvailable name = "@1@"
+            ]
+            stackinggroup = "WEAPON_BASE_LOCKED_EFFECTS_@1@"
+            accountinglabel = "@1@"
+            effects = [
+                SetMaxCapacity partname = "@1@" value = Value + PartCapacity name = "@1@"
+                SetMaxSecondaryStat partname = "@1@" value = Value + PartSecondaryStat name = "@1@"
+            ]
+'''
```
Comment (Morlic-fo) [inline default/scripting/ship_parts/Weapons.focs.txt]: The effect is now the same as the unowned macro. Seems like merging them into one macro (activation = Unowned or Not OwnerHasShipPartAvailable) would make more sense.

Comment (Dilvish-fo) [reply default/scripting/ship_parts/Weapons.focs.txt]: > Seems like merging them into one macro (activation = Unowned or Not OwnerHasShipPartAvailable) would make more sense.

Although in the current uses it seems we are always pairing these together, it does not strike me that we would necessarily *always* want to do so; I'd be inclined to leave them as separate things as the PR currently has them.

Comment (dbenage-cx) [reply default/scripting/ship_parts/Weapons.focs.txt]: I've merged to the one macro in the interest of moving forward.

**Comment by Vezzra** [comment]
Everyone, especially @dbenage-cx: status report? Anything preventing this from being merged?

**Comment by dbenage-cx** [comment]
Will merge tomorrow if no objections.

**Pull request merged**